src/EditorFeatures/Core/IntelliSense/AsyncCompletion/ItemManager.CompletionListUpdater.cs

```diff
@@ -128,20 +132,24 @@ private bool ShouldSelectSuggestionItemWhenNoItemMatchesFilterText
                 // since the completion list could be long with import completion enabled.
                 var itemsToBeIncluded = s_listOfMatchResultPool.Allocate();
                 var cancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+                var threadLocalPatternMatchHelper = new ThreadLocal<PatternMatchHelper>(() => new PatternMatchHelper(_filterText), trackAllValues: true);
```

Comment:
what does 'trackAllValues' do?

Reply:
looking at the docs, unclear why we're setting that to true.  can you doc in the code why?

also, doc in the code why we're using a ThreadLocal here (specifically that PatternMatchHelper is not threadsafe, so we want a dedicated one per thread that is active doing the filtering.